### PR TITLE
Fix #15735: Glossary,Term, Classification,Tag rename operation fixes

### LINKED ISSUE #15735: Unable to approve a glossary Term
**Affected module**
UI

**Describe the bug**
After creating a glossary term (don't aprove it yet), change it's name. The task shows one task, but the ui doesn't display it. So we're unable to approve the glossary term

**To Reproduce**

image

image

**Expected behavior**
To be able to approve.

**Version:**
 - OpenMetadata version: 1.3.1

**Additional context**

https://sandbox.open-metadata.org/glossary/Gloss%C3%A1rio%20Comercial.Folha%20de%20Or%C3%A7amento%20de%20Projeto%20(FOP)/activity_feed/tasks

### LINKED ISSUE #16853: Create/Edit(udpate) Glossary field Description*  allow input only spacebar ?
**Affected module**
Create new glossary/ Edit glossary 

create_description_only_spacebar

**Describe the bug**
Create new and edit glossary field Description allow input only spacebar 
**To Reproduce**

allow_input_spacebar

- Click button add glossary
- Fill data in field
- Input field Description only spacebar 
- Save

**Expected behavior**
Trim description field .
- Not allow input only spacebar

**Version:**
  - OpenMetadata version: 1.4.3
 
**Additional context**
i want custom openmetada in localhost . please support build code and test in local ( clip , guide ..... ) 
thank alot.

### PR DESCRIPTION
**Pull request by sonika-shah**
### Describe your changes:

Fixes #15735 #12700 
Fixes #16853 

https://github.com/open-metadata/OpenMetadata/assets/33024356/ebb0397f-bc93-4d49-aa27-a14c710cf272

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/ClassificationRepository.java [hunk 1]
```diff
@@ -112,6 +117,39 @@ public TagLabel map(ResultSet r, org.jdbi.v3.core.statement.StatementContext ctx
     }
   }
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/ClassificationRepository.java:119]: @sonika-shah whenever tag rename or glossary term rename happens. We should do this bulk operation in asynchronous fashion not in-line.
Might as well take the quartz approach and put a job that can handle failures and retries

Comment (sonika-shah) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/ClassificationRepository.java:119]: yes @harshach , the reindexing done via `reindexAcrossIndices`  is happening in [async](https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java#L1529C1-L1530C1) fashion

**Review by Sachin-chaurasiya — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/playwright/utils/glossary.ts [hunk 2]
```diff
@@ -590,3 +594,106 @@ export const createGlossaryTerms = async (
...
+
+  await page.click('[data-testid="glossary-term-add-button-menu"]');
+  await page
+    .locator('.ant-dropdown-menu .ant-dropdown-menu-title-content')
+    .filter({ hasText: 'Assets' })
+    .click();
+
+  await page
+    .locator('[data-testid="asset-selection-modal"] .ant-modal-title')
+    .filter({ hasText: 'Add Assets' });
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/playwright/utils/glossary.ts:623]: this is not doing anything!

File: openmetadata-ui/src/main/resources/ui/playwright/support/glossary/GlossaryTerm.ts [hunk 3]
```diff
@@ -138,4 +138,9 @@ export class GlossaryTerm {
 
     return await response.json();
   }
+
+  rename(newTermName: string, newTermFqn: string) {
+    this.responseData.name = newTermName;
+    this.responseData.fullyQualifiedName = newTermFqn;
+  }
```
[the remark this answers is not in the document]
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/playwright/support/glossary/GlossaryTerm.ts:145]: This is required for further delete operation.
In glossary term we allow rename of `name` field, in that case name and fullyqualifiedName is updated.
Need to update the term object in that case.

**Review by ShaileshParmar11 — DISMISSED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java [hunk 4]
```diff
@@ -406,6 +406,15 @@ protected void restorePatchAttributes(T original, T updated) {
     updated.setChangeDescription(original.getChangeDescription());
   }
 
+  /**
+   * This function updates the Elasticsearch indexes wherever the specific entity is present.
+   * It is typically invoked when there are changes in the entity that might affect its indexing in Elasticsearch.
+   * The function ensures that the indexes are kept up-to-date with the latest state of the entity across all relevant Elasticsearch indexes.
+   */
+  protected void reindexIndexesOnEntitySpecificUpdate(T original, T updated) {
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java:414]: entitySpecificReindex()

Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java:414]: entityRelationshipReindex()

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java [hunk 5]
```diff
@@ -869,6 +878,9 @@ public final PatchResponse<T> patch(UriInfo uriInfo, UUID id, String user, JsonP
     // Update the attributes and relationships of an entity
     EntityUpdater entityUpdater = getUpdater(original, updated, Operation.PATCH);
     entityUpdater.update();
+
+    reindexIndexesOnEntitySpecificUpdate(original, updated);
```
Comment (mohityadav766) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java:882]: also should this only be done when there is ENTITY CHANGE from update()

**Review by chirag-madlani — APPROVED** [review]

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**